qiskit/synthesis/linear_phase/cx_cz_depth_lnn.py

```diff
@@ -245,6 +245,7 @@ def synth_cx_cz_depth_line_my(mat_x: np.ndarray, mat_z: np.ndarray) -> QuantumCi
     n = len(mat_x)
     mat_x = calc_inverse_matrix(mat_x)
 
+    mat_x = mat_x.astype(bool)
```

Comment:
Same comment as before `calc_inverse_matrix` is returning a dtype bool array. Do you need to copy this here? If so I would do that from rust (especially since this would be a pattern of usage). But either way you can drop this line

```suggestion
````